src/Sylius/Component/Core/spec/Uploader/ImageUploaderSpec.php

```diff
@@ -16,37 +16,46 @@
 use Gaufrette\Filesystem;
 use PhpSpec\ObjectBehavior;
 use Prophecy\Argument;
+use Sylius\Component\Core\Generator\ImagePathGeneratorInterface;
 use Sylius\Component\Core\Model\ImageInterface;
 use Sylius\Component\Core\Uploader\ImageUploaderInterface;
 use Symfony\Component\HttpFoundation\File\File;
 
 final class ImageUploaderSpec extends ObjectBehavior
 {
-    function let(Filesystem $filesystem, ImageInterface $image): void
-    {
+    function let(
+        Filesystem $filesystem,
+        ImagePathGeneratorInterface $imagePathGenerator,
+        ImageInterface $image
+    ): void {
         $filesystem->has(Argument::any())->willReturn(false);
 
         $file = new File(__FILE__);
         $image->getFile()->willReturn($file);
 
-        $this->beConstructedWith($filesystem);
+        $this->beConstructedWith($filesystem, $imagePathGenerator);
```

Comment:
Could we add a test for the deprecation (`$this->shouldTrigger(\E_USER_DEPRECATED)->during...()` matcher).